### PR TITLE
Jz

### PR DESCRIPTION
**Pull request by JunfeiZhang**

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -71,9 +71,7 @@ function createGraphReport(sitemon: any, date: string): void {
     createChart(option);
 }
 
-function createChartOption(
-    chartUpdateOption: ChartUpdateOption
-): echarts.EChartOption {
+function createChartOption(chartUpdateOption: ChartUpdateOption): echarts.EChartOption {
```
Comment (zicodeng) [inline src/options.ts:74]: min diff

File: src/background.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ const chromep = new ChromePromise();
 
 // A global Timer that tracks how long the current tab has been active (in seconds).
 let timer = 0;
-setInterval(() => {
+let intervalId:NodeJS.Timer|null = setInterval(() => {
```
Comment (zicodeng) [inline src/background.ts]: Go to vscode and download `prettier` plugin. This plugin will automatically format it for you so the team can have consistent style

File: src/background.ts [hunk 3]
```diff
@@ -20,11 +20,27 @@ let currentHostname: string;
...
         listenToTabStatusChange();
         listenToRuntimeMessages();
+        listenToIdleTimer();
     } catch (error) {
         console.error(error);
     }
 })();
 
+function listenToIdleTimer(){
+    chrome.idle.setDetectionInterval(15);
```
Comment (zicodeng) [inline src/background.ts]: Maybe write a comment here so that we know what time units here is?

File: src/background.ts [hunk 4]
```diff
@@ -20,11 +20,27 @@ let currentHostname: string;
...
+
+    chrome.idle.onStateChanged.addListener((newState) => {
```
Comment (zicodeng) [inline src/background.ts]: Specify type of newState

File: src/background.ts [hunk 5]
```diff
@@ -20,11 +20,27 @@ let currentHostname: string;
...
+        if (newState != 'active' && intervalId){
+            clearInterval(intervalId);
+            intervalId = null;
+        } else {
+            intervalId = setInterval(() => {
```
Comment (zicodeng) [inline src/background.ts]: Maybe create a function called `startTimer` to avoid DRY

File: src/background.ts (hunk 2 above)
Comment (zicodeng) [inline src/background.ts]: How about `timerId`?

File: src/background.ts [hunk 6]
```diff
@@ -20,11 +20,27 @@ let currentHostname: string;
...
     }
 })();
 
+function listenToIdleTimer(){
+    chrome.idle.setDetectionInterval(15);
+
+    chrome.idle.onStateChanged.addListener((newState) => {
+        if (newState != 'active' && intervalId){
+            clearInterval(intervalId);
+            intervalId = null;
```
Comment (zicodeng) [inline src/background.ts]: Why set it to null??

**Review by zicodeng — COMMENTED** [review]
Great job implementing this logic. Some inconsistent style issue which can be fixed by implementing prettier plugin.

File: src/background.ts (hunk 6 above)
Comment (JunfeiZhang) [reply src/background.ts]: clearInterval() with invalid ID will clear all intervals. To prevent this, we reset intervalId.

**Pull request merged**